Core/Core/Api/GraphQL/Resources/CommentResource.cs

```diff
@@ -0,0 +1,107 @@
+﻿using System.Threading;
+using System.Threading.Tasks;
+using GraphQL;
+using Speckle.Core.Api.GraphQL.Inputs;
+using Speckle.Core.Api.GraphQL.Models;
+using Speckle.Core.Api.GraphQL.Models.Responses;
+
+namespace Speckle.Core.Api.GraphQL.Resources;
+
+public sealed class CommentResource
+{
+  private readonly ISpeckleGraphQLClient _client;
+
+  internal CommentResource(ISpeckleGraphQLClient client)
+  {
+    _client = client;
+  }
+
+  /// <param name="projectId"></param>
+  /// <param name="limit">Max number of comments to fetch</param>
+  /// <param name="cursor">Optional cursor for pagination</param>
+  /// <param name="filter">Optional filter</param>
+  /// <param name="repliesLimit">Max number of comment replies to fetch</param>
+  /// <param name="repliesCursor">Optional cursor for pagination</param>
+  /// <param name="cancellationToken"></param>
+  /// <returns></returns>
+  /// <inheritdoc cref="ISpeckleGraphQLClient.ExecuteGraphQLRequest{T}"/>
+  public async Task<ResourceCollection<Comment>> GetProjectComments(
+    string projectId,
+    int limit = 25,
+    string? cursor = null,
+    ProjectCommentsFilter? filter = null,
+    int repliesLimit = 25,
+    string? repliesCursor = null,
+    CancellationToken cancellationToken = default
```

Comment:
I really would like the option to inject these tbh so we don't have to keep passing it around...

Reply:
not for now...

Reply:
Inject what? the defaults?

The issue is limits should be hand crafted for each caller based on what is sensible to them, its not something that should be abstracted away from the call site, same for the cursors and filters...

The 25 value is also the default limit defined in the schema, so unless you somehow want to pull this directly from the schema, I'm happy to simply make it a `const int` in a file somewhere

Reply:
The cancellation token, we seem to be passing it about but I think there could be ways to just pick up the token that's been injected in a lot of cases. It might only work within a UOW though atm